src/persist/src/sqlite.rs

```diff
@@ -208,6 +182,76 @@ impl Consensus for SqliteConsensus {
             Ok(Err(current))
         }
     }
+
+    async fn scan(
+        &self,
+        _deadline: Instant,
+        key: &str,
+        from: SeqNo,
+    ) -> Result<Vec<VersionedData>, ExternalError> {
+        let conn = self.conn.lock().await;
+        let mut stmt = conn.prepare_cached(
+            "SELECT sequence_number, data FROM consensus
+                 WHERE shard = $shard AND sequence_number >= $from
+                 ORDER BY sequence_number",
+        )?;
+        let rows = stmt.query_map(named_params! {"$shard": key, "$from": from}, |row| {
+            let seqno = row.get("sequence_number")?;
+            let data: Vec<_> = row.get("data")?;
+            Ok(VersionedData { seqno, data })
+        })?;
+
+        rows.collect::<Result<Vec<_>, _>>().map_err(|e| e.into())
+    }
+
+    async fn truncate(
+        &self,
+        deadline: Instant,
+        key: &str,
+        seqno: SeqNo,
+    ) -> Result<Result<(), Option<VersionedData>>, ExternalError> {
+        let result = {
+            let conn = self.conn.lock().await;
+            let mut stmt = conn.prepare_cached(
+            "DELETE FROM consensus
+             WHERE shard = $shard AND sequence_number < $sequence_number AND
+             EXISTS(
+                 SELECT * FROM consensus WHERE shard = $shard AND sequence_number >= $sequence_number
+             )",
+        )?;
```

Comment:
nit: the current indentation is misleading. I think we can push this up a line (by deleting the comma that trails the string)

Reply:
i removed the comma but that didn't do anything?